### ISSUE TITLE
spire-agent helm chart causes unnecessary warnings

### ISSUE THREAD

**Issue by gflarity** [spiffe/helm-charts-hardened#396]
Out of the box, when you install the spire helm chart which  includes the spire-agent helm chart, you'll get periodic warnings  in the agent logs similar to:

```
time="2024-07-08T03:20:29+02:00" level=error msg="Invalid argument: failed to attest: join token does not exist or has already been used" authorized_as=nobody authorized_via= caller_addr="100.101.102.103:50744" method=AttestAgent node_attestor_type=join_token request_id=57131fe3-58e4-4700-b089-ca250ebf896e service=agent.v1.Agent subsystem_name=api
```

This results in the agent's svid not being persisted and unnecessary re-attestation during start up.  Not a huge deal, but it also seems like this is easy to fix.  

Looks like this because the config map sets data_dir to ["/run/spire"](https://github.com/spiffe/helm-charts-hardened/blob/9c77280e6ea8e07c5e17ec0a49a5697ac1eda292/charts/spire/charts/spire-agent/templates/configmap.yaml#L38) however [the daemonset is doesn't create this volume/mount](https://github.com/spiffe/helm-charts-hardened/blob/9c77280e6ea8e07c5e17ec0a49a5697ac1eda292/charts/spire/charts/spire-agent/templates/daemonset.yaml#L190).  

It seems like just adding a volume/mount of spire-data-dir which points to /run/spire would resolve this issue?

I'm willing to submit a PR, just wanted to make sure the above sounds accurate and would likely get merged first.

**Comment by kfox1111** [comment]
Hi @gflarity. Thanks for reporting the issue.

We're only supporting k8s and tpm node attestors currently, and it looks like both support reattestation, so we never noticed it isn't storing that file, as it works ok without it.

I think we probably need to discuss options in the next sync.

There are some advantages to not having a hostpath mount when not needed. So maybe should be an optional feature.

Where on the host should the data be placed by default is also another concern.

**Comment by kfox1111** [comment]
This did not autoclose fixed along with #414. Closing manually.
